rust-toolchain

```diff
@@ -1 +1,3 @@
-nightly-2020-11-24
+[toolchain]
```

Comment:
I've come up with this trick:

```suggestion
# If you see this, run `rustup self update` to get rustup 1.23 or newer.

# NOTE: above comment is for older `rustup` (before TOML support was added),
# which will treat the first line as the toolchain name, and therefore show it
# to the user in the error, instead of "error: invalid channel name '[toolchain]'".

[toolchain]
```

The result of that is that if you're on older `rustup`, you get this error:
```
error: invalid channel name '# If you see this, run `rustup self update` to get rustup 1.23 or newer.' in '/home/eddy/Projects/rust-gpu/rust-toolchain'
error: caused by: invalid toolchain name: '# If you see this, run `rustup self update` to get rustup 1.23 or newer.'
```
instead of the much less informative:
```
error: invalid channel name '[toolchain]' in '/home/eddy/Projects/rust-gpu/rust-toolchain'
error: caused by: invalid toolchain name: '[toolchain]'
```